File: 7-kyu/easy-wallpaper/c++/solution.cpp
#include <cmath>
#include <vector>

using namespace std;

class Wallpaper
{
  public:
  static std::string wallPaper(double l, double w, double h);
};

std::string Wallpaper::wallPaper(double l, double w, double h)
{
    const double roll_size = 5.2;
    int rolls_required =
    	std::ceil(((2 * w * h) + (2 * h * l)) * 1.15 / roll_size);
    std::vector<std::string> words = { "zero", "one", "two", "three", "four",
    	"five", "six", "seven", "eight", "nine", "ten", "eleven", "twelve",
    	"thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen",
    	"nineteen", "twenty" };
    return words.at(rolls_required);
}

File: 7-kyu/easy-wallpaper/swift/solution.swift


func wallpaper(_ l: Double, _  w: Double,_  h: Double) -> String {
    let numbers = ["zero", "one", "two", "three", "four", "five",
               "six", "seven", "eight", "nine", "ten",
               "eleven", "twelve", "thirteen", "fourteen", "fifteen",
               "sixteen", "seventeen", "eighteen", "nineteen", "twenty"]
    guard l > 0.0 && w > 0.0 && h > 0.0 else { return numbers[0] }
    var result: Double = ((2.0 * w * h) + (2.0 * h * l)) * 1.15 / 5.2
    result.round(.up)
    let index = Int(result)
    return numbers[index]
}